### ISSUE TITLE
NaN values in las files

### ISSUE THREAD

**Issue by tiedl25** [CloudCompare/CloudCompare#1695]
**Describe the bug**
Since version 2.12.1 NaN values in scalar fields are automatically converted to a default value when saving to a las file. This is problematic for applications/scripts that rely on those values. There should be at least an option to set the default value manually. 

**How to reproduce**

**Expected behaviour**
An option to set the default value manually (also in commandline mode)

**Additional context**

**Your environment**
 - CC Version (or git hash/tag): 2.12.4
 - OS & Version: Windows 10

**Comment by dgirardeau** [comment]
Do you mean that some applications rely on NaN values?

Because it didn't seem that LAS specifications allowed NaN values (and some SW tools reject LAS files with NaN values). That's why we automatically convert them now. Do you know if there's an official way to manage NaN values with LAS files maybe?

And if not, I believe that to 'solve' your issue we would probably prefer a more generic way to change NaN values before saving to LAS or any other format. For instance a special operation of the SF Arithmetic tool.

**Comment by tiedl25** [comment]
https://github.com/ASPRSorg/LAS/wiki/Standard-ExtraByte-Definitions#published-standard-extrabyte-definitions
> As a general guideline, the maximum possible value is a good default NODATA value, except in the cases when zero has the same meaning as null and is therefore preferable.

For example the M3C2-distances parameter sometimes contains nan values but also zeros, which do not have the same meaning and shouldn't be treated the same way.
So it would be great to have an option in commandline mode to change this default value or leave it as NaN.

**Comment by dgirardeau** [comment]
Yes, but what would be the default value of NODATA in a general case? (as I see that it depends on the application, and quite often '0' in the examples...).

**Comment by tiedl25** [comment]
Agreed, there is no one-for-all (applications) general case. Therefore, the NoData value should be determined based on the value range of each attribute, e.g. > maximum possible value, as documented here: https://github.com/ASPRSorg/LAS/wiki/Standard-ExtraByte-Definitions#published-standard-extrabyte-definitions.
Looking at the LAS standard, there is a NoData flag available for extra byte attributes, which should be used to store the NoData value (Table 25 in this document: http://www.asprs.org/wp-content/uploads/2019/03/LAS_1_4_r14.pdf).
When reading LAS files, the NoData flag must be checked and all values occurring in the corresponding attribute are to be considered NoData/nan.
